xarray/backends/api.py

```diff
@@ -455,43 +462,25 @@ def maybe_decode_store(store, lock=False):
 
         _protect_dataset_variables_inplace(ds, cache)
 
-        if chunks is not None:
-            from dask.base import tokenize
-
-            # if passed an actual file path, augment the token with
-            # the file modification time
-            if isinstance(filename_or_obj, str) and not is_remote_uri(filename_or_obj):
-                mtime = os.path.getmtime(filename_or_obj)
-            else:
-                mtime = None
-            token = tokenize(
-                filename_or_obj,
-                mtime,
-                group,
-                decode_cf,
-                mask_and_scale,
-                decode_times,
-                concat_characters,
-                decode_coords,
-                engine,
-                chunks,
-                drop_variables,
-                use_cftime,
-            )
-            name_prefix = "open_dataset-%s" % token
-            ds2 = ds.chunk(chunks, name_prefix=name_prefix, token=token)
-            ds2._file_obj = ds._file_obj
-        else:
-            ds2 = ds
-
-        return ds2
+        return ds
 
     if isinstance(filename_or_obj, Path):
         filename_or_obj = str(filename_or_obj)
 
     if isinstance(filename_or_obj, AbstractDataStore):
         store = filename_or_obj
 
+    if isinstance(filename_or_obj, MutableMapping):
+        if engine == 'zarr':
+            # on ZarrStore, mode='r', synchronizer=None, group=None,
+            # consolidated=False.
+            overwrite_encoded_chunks = backend_kwargs.pop("overwrite_encoded_chunks", None)
```

Comment:
`pop` modifies a dict inplace. It's best to make a copy of `backend_kwargs` first to ensure that objects users supply as arguments into xarray functions aren't unexpected modified.